MarkBind/markbind
# Finalize migration to Bootstrap 4.0

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, please explain: Migration


Fixes #316.



**What is the rationale for this request?**
We need to migrate to Bootstrap 4 if we want to utilize bootstrap-vue.

**What changes did you make? (Give an overview)**
* Upgraded Bootstrap to v4, and installed Bootstrap-Vue.
* Font-Awesome was imported, and glyphicons revived.
* Documentation were updated.
* Website was updated to fix breaking changes.

**Provide some example code that this change will affect:**
\-

**Is there anything you'd like reviewers to focus on?**
Testing that nothing breaks.

**Testing instructions:**
1. Upgrade to incoming vue-strap v2.0.1-markbind.14, by building the [`bootstrap-migration` branch](https://github.com/MarkBind/vue-strap/pull/64), copying `dist/vue-strap.min.js` from vue-strap repository to MarkBind repository (`assets/js`).
2. Clone `https://github.com/nus-cs2103/website-base.git`.
3. Do the following updates to your `website-base` to fix the breaking changes:
  a. Update the navbar to use the new classes (https://github.com/yamgent/website-base/commit/e7a801c5993e3bf4ddafd4086b8458f1c20d5a3b).
  b. Change the html tag for site-nav from `<markdown>` to `<navigation>` (https://github.com/yamgent/website-base/commit/d7cc0e0084ffa1c6ae7fcef5ee1df4d60b52023c).
4. Run markbind serve. No errors should occur.
5. Eye-ball anything to ensure that nothing is broken.
